### PR TITLE
fix: NullReferenceException when referrers index has null manifests

### LINKED ISSUE #398: FetchReferrersByApi throws NullReferenceException when registry returns "manifests": null
### Description

When listing referrers via the Referrers API, oras-dotnet crashes with a `NullReferenceException` if the registry returns `"manifests": null` instead of an empty array `"manifests": []`.

Per the [OCI Distribution Spec v1.1.1](https://github.com/opencontainers/distribution-spec/blob/v1.1.1/spec.md#listing-referrers), a registry with no referrers should return an empty `manifests` array. However, some registries (e.g. `dhi.io`) return `null` for the `manifests` field. This is technically a registry-side bug, but `oras` (Go) and `oras-go` tolerate it gracefully, so oras-dotnet should too.

### Root cause

`Oci.Index.Manifests` is declared as a non-nullable `required IList<Descriptor>`:

```csharp
[JsonPropertyName("manifests")]
public required IList<Descriptor> Manifests { get; set; }
```

When the JSON contains `"manifests": null`, `System.Text.Json` sets the property to `null`. Code in `Repository.FetchReferrersByApi` (and `PullReferrersIndexList`) then iterates `referrersIndex.Manifests`, throwing `NullReferenceException`.

### Why oras / oras-go are unaffected

oras-go decodes into `ocispec.Index`, where `Manifests` is a Go slice. A JSON `null` becomes a `nil` slice, and both `len(nil)` and `range nil` are safe no-ops. The `oras` CLI (`cp`, etc.) inherits this behavior — e.g. `cmd/oras/root/cp.go` does `if len(index.Manifests) == 0`, which is null-safe.

### Steps to reproduce

1. Point a copy/referrers operation at a repository on a registry that returns `"manifests": null` for a subject with no referrers (e.g. `dhi.io`).
2. Observe `NullReferenceException` during referrers enumeration.

### Proposed fix

Coalesce a `null` `manifests` value to an empty list in the `Index` model so it is never null after deserialization, mirroring Go's nil-slice semantics. Add tests covering deserialization of `"manifests": null` and an end-to-end `FetchReferrersByApi` test with a null-manifests response.

### Affected specs / references
- OCI Distribution Spec v1.1.1 – Listing Referrers
- Mirrors `oras-go` `registry/remote/repository.go` `referrersPageByAPI`

### PR DESCRIPTION
**Pull request by renshao**
## What this PR does

Fixes a `NullReferenceException` that occurs when a registry returns `"manifests": null` (instead of an empty array `"manifests": []`) for a subject with no referrers.

Per the [OCI Distribution Spec v1.1.1](https://github.com/opencontainers/distribution-spec/blob/v1.1.1/spec.md#listing-referrers), a registry with no referrers should return an empty `manifests` array. However, some registries (e.g. `dhi.io`) return `null`. `oras` (Go) and `oras-go` tolerate this because a JSON `null` decodes to a `nil` slice, and `len(nil)`/`range nil` are safe no-ops. oras-dotnet's `Index.Manifests` is a non-nullable `required IList<Descriptor>`, so `System.Text.Json` set it to `null`, and enumerating it in `FetchReferrersByApi`/`PullReferrersIndexList` threw.

## Change

- `Oci.Index.Manifests` now coalesces a `null` value to an empty list via a backing field, so it is never null after deserialization — mirroring Go's nil-slice semantics.

## Tests

- `Deserialize_NullManifests_CoalescesToEmptyList` and a new `"manifests": null` fixture in the index serialization tests.
- `Repository_ReferrersByApi_NullManifests_ReturnsEmpty` — end-to-end test where the mocked Referrers API returns `"manifests": null`.

All 555 tests pass.

Fixes #398

### CODE REVIEW

File: src/OrasProject.Oras/Oci/Index.cs [hunk 1]
```diff
@@ -34,8 +34,17 @@ public class Index : Versioned
     public string? ArtifactType { get; set; }
 
     // Manifests references platform specific manifests.
+    // A null JSON value (e.g. `"manifests": null`) is coalesced to an empty
+    // list to mirror Go's nil-slice semantics and tolerate non-conformant
```
Comment (sajayantony) [inline src/OrasProject.Oras/Oci/Index.cs:38]: Why do we say that null is non-conformant. I don't think the image spec calls this out.

Comment (renshao) [reply src/OrasProject.Oras/Oci/Index.cs:38]: See this section: https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers

>  If a query results in no matching referrers, an empty manifest list MUST be returned.

I agree that the term 'empty manifest list' can be interpreted differently. Regardless, there's no harm oras-dotnet handles both `[]` and `null`, this also matches oras-go's behavior.

**Review by akashsinghal — DISMISSED** [review]

**Review by akashsinghal — COMMENTED** [review]
@renshao. @sajayantony had a valid concern he mentioned offline: "I'm considering this a bit more in terms of an SDK author. Let's say we did want to produce manifest with null, now we can't since we dont' allow the value setting. There is nothing in the OCI spec that say that this value cannot be null and if we accept that and reserialize it it should not produce []. 

So basically I'm thinking that the fix should be in the consumer side and not on the property.
 
My thinking is that we should fix FetchReferrersByApi

**Comment by akashsinghal** [comment]
closing in favor of #408
